catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ServerAuthenticationMode.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.catalog.files.s3;
+
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ContainerCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.InstanceProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.SystemPropertyCredentialsProvider;
+
+/**
+ * @see <a
+ *     href="https://github.com/quarkiverse/quarkus-amazon-services/blob/4ecc71767857d476767ffd36f0fac98c1b2b7de1/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/AwsCredentialsProviderType.java">Quarkus
+ *     AWS AwsCredentialsProviderType</a>
+ */
+public enum S3ServerAuthenticationMode {
+  DEFAULT {
+    @Override
+    public AwsCredentialsProvider newCredentialsProvider(S3BucketOptions bucketOptions) {
+      return DefaultCredentialsProvider.builder()
+          .asyncCredentialUpdateEnabled(false)
+          .reuseLastProviderEnabled(true)
+          .build();
+    }
+  },
+
+  STATIC {
+    @Override
+    public AwsCredentialsProvider newCredentialsProvider(S3BucketOptions bucketOptions) {
+      return bucketOptions
+          .accessKey()
+          // TODO support session token: AwsSessionCredentials.create(accessKeyId, secretAccessKey,
+          // sessionToken)
```

Comment:
Should I provide support for this in this PR? This is supported in Quarkus. I guess in this case, the STS token is obtained prior to starting the server and assumed to be valid for long enough, "à la" bearer token auth.